src/Umbraco.Core/Services/Implement/FileService.cs

```diff
@@ -405,6 +405,11 @@ public long GetScriptFileSize(string filepath)
         /// <returns></returns>
         public ITemplate CreateTemplateWithIdentity(string name, string alias, string content, ITemplate masterTemplate = null, int userId = Constants.Security.SuperUserId)
         {
+            if (name != null && name.Length > 255)
+            {
+                throw new InvalidOperationException("Name cannot be more than 255 characters in length.");
+            }
```

Comment:
So this check should be separated into:
```c#
if (name == null) throw new ArgumentNullException(nameof(name));
if (string.IsNullOrWhiteSpace(name)) throw new ArgumentException("Value can't be empty or consist only of white-space characters.", nameof(name));
if (name.Length > 255) throw new ArgumentOutOfRangeException(nameof(name), "Value can't be more than 255 characters in length.");
```

Reply:
Thanks @ronaldbarendse , working on it now :-)